connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSinkTaskTest.java

```diff
@@ -297,15 +298,21 @@ public void testErrorInRebalancePartitionAssignment() throws Exception {
     @Test
     public void testWakeupInCommitSyncCausesRetry() throws Exception {
         expectInitializeTask();
-        expectPollInitialAssignment();
 
-        final List<TopicPartition> partitions = asList(TOPIC_PARTITION, TOPIC_PARTITION2);
+        expectPollInitialAssignment();
 
-        sinkTask.close(new HashSet<>(partitions));
+        expectConsumerPoll(1);
```

Comment:
Needed to add a new poll that returns a message, otherwise an offset commit was not triggered due to the new check that skips offset commit if the offsets-to-commit are identical to the workerTask's `lastCommittedOffsets`.